integration-tests/client/test/core.test.ts

```diff
@@ -0,0 +1,355 @@
+            };
+        }
+    }
+
+    console.log("waiting for stride to start...");
+    await waitForChain(strideAccounts.user, USTRD);
+
+    console.log("waiting for gaia to start...");
+    await waitForChain(gaiaAccounts.user, UATOM);
+
+    console.log("waiting for stride-gaia ibc...");
+    await waitForIbc(
+        strideAccounts.user,
+        TRANSFER_CHANNEL.STRIDE.GAIA!,
+        USTRD,
+        "cosmos",
+    );
+}, 45_000);
+
+describe("Core Tests", () => {
+    test("IBC Transfer", async () => {
+        const stridejs = strideAccounts.user;
+        const gaiajs = gaiaAccounts.user;
+        const transferAmount = 50000000;
+
+        // Get initial balances - only track what we can reliably query
+        const { balance: { amount: strideInitialStrideBalance } = { amount: "0" } } =
+            await stridejs.query.cosmos.bank.v1beta1.balance({
+                address: stridejs.address,
+                denom: USTRD,
+            });
+
+        const { balance: { amount: strideInitialAtomBalance } = { amount: "0" } } =
+            await stridejs.query.cosmos.bank.v1beta1.balance({
+                address: stridejs.address,
+                denom: ATOM_DENOM_ON_STRIDE,
+            });
+
+        const gaiaInitialAtomBalance = await gaiajs.query.bank.balance(gaiajs.address, UATOM);
+
+        console.log("Initial balances:");
+        console.log(`Stride USTRD: ${strideInitialStrideBalance}`);
+        console.log(`Stride ATOM: ${strideInitialAtomBalance}`);
+        console.log(`Gaia ATOM: ${gaiaInitialAtomBalance.amount}`);
+
+        // Perform IBC transfers
+        console.log("Transferring USTRD from Stride to Gaia...");
+        await ibcTransfer({
+            client: stridejs,
+            sourceChain: "STRIDE",
+            destinationChain: "GAIA",
+            coin: `${transferAmount}${USTRD}`,
+            sender: stridejs.address,
+            receiver: gaiajs.address,
+        });
+
+        console.log("Transferring ATOM from Gaia to Stride...");
+        await ibcTransfer({
+            client: gaiajs,
+            sourceChain: "GAIA",
+            destinationChain: "STRIDE",
+            coin: `${transferAmount}${UATOM}`,
+            sender: gaiajs.address,
+            receiver: stridejs.address,
+        });
+
+        // Wait a bit for transfers to complete
+        await sleep(5000);
+
+        // Get final balances
+        const { balance: { amount: strideFinalStrideBalance } = { amount: "0" } } =
+            await stridejs.query.cosmos.bank.v1beta1.balance({
+                address: stridejs.address,
+                denom: USTRD,
+            });
+
+        const { balance: { amount: strideFinalAtomBalance } = { amount: "0" } } =
+            await stridejs.query.cosmos.bank.v1beta1.balance({
+                address: stridejs.address,
+                denom: ATOM_DENOM_ON_STRIDE,
+            });
+
+        const gaiaFinalAtomBalance = await gaiajs.query.bank.balance(gaiajs.address, UATOM);
```

Comment:
same here with gaia final strd balance

Reply:
also can you similarly unpack this so that it's an int instead of  a struct?

Reply:
addressed